### PR TITLE
Doc: Extending streaming/long-running sub-resource

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: docs/extending_streaming_subresource [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Extending Streaming Sub-Resource
+
+**TL;DR**: This document will be helpful if you're trying to extend/customize a streaming long-running sub-resource for 
+your aggregated-apiservers, similar to `pods/logs`. `pods/exec` endpoints in kube-apiserver. Before you diving into
```
Comment (wallrj) [inline docs/extending_streaming_subresource]: typo "Before diving into it"

File: docs/extending_streaming_subresource [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Extending Streaming Sub-Resource
+
+**TL;DR**: This document will be helpful if you're trying to extend/customize a streaming long-running sub-resource for 
+your aggregated-apiservers, similar to `pods/logs`. `pods/exec` endpoints in kube-apiserver. Before you diving into 
+it, you're suppose to know not only [how to create a subresource](https://github.com/kubernetes-incubator/apiserver-builder-alpha/blob/master/docs/adding_subresources.md#adding-a-subresource-to-a-resource)
```
Comment (wallrj) [inline docs/extending_streaming_subresource]: typo "supposed"

File: docs/extending_streaming_subresource [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Extending Streaming Sub-Resource
+
+**TL;DR**: This document will be helpful if you're trying to extend/customize a streaming long-running sub-resource for 
+your aggregated-apiservers, similar to `pods/logs`. `pods/exec` endpoints in kube-apiserver. Before you diving into 
+it, you're suppose to know not only [how to create a subresource](https://github.com/kubernetes-incubator/apiserver-builder-alpha/blob/master/docs/adding_subresources.md#adding-a-subresource-to-a-resource) 
+but also [how to add custom REST](https://github.com/kubernetes-incubator/apiserver-builder-alpha/blob/master/docs/adding_custom_rest.md) 
+for your aggregated apiserver. By reading it, you will basically learn to plumb long-running sub-resources in two ways.
```
Comment (wallrj) [inline docs/extending_streaming_subresource]: Drop "basically"

File: docs/extending_streaming_subresource [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+### What is streaming sub-resource?
```
Comment (wallrj) [inline docs/extending_streaming_subresource]: typo "What is *a* steaming sub-resource"

File: docs/extending_streaming_subresource [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+Basically, kubernetes provides a few built-in support for basic CRUD verbs over resource or sub-resource, including
```
Comment (wallrj) [inline docs/extending_streaming_subresource]: Drop "Basically"

Comment (wallrj) [reply docs/extending_streaming_subresource]: Capitalize Kubernetes

Comment (wallrj) [reply docs/extending_streaming_subresource]: Drop "a few"

**Comment by yue9944882** [comment]
@wallrj thanks, when i was trying to append more content to this WIP documentation, i found that the implementation of `pod/log` from upstream isn't reasonably designed so i paused the work. it can actually converge w/ other implementations of streaming subresources like `pod/exec`, i opened #79430 to do the experiment and it seems to be working.. if you're looking for how to extend streaming subresources, there're examples at https://github.com/kubernetes-sigs/apiserver-builder-alpha#examples.

**Pull request closed without merging**